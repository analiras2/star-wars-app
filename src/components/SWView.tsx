import React, { PureComponent } from 'react';
import { View } from 'react-native';
import styles from '../res/styles';

interface Props {
  children: Element;
  row?: boolean;
  flex?: number;
  bgColor?: string;
  align?:
    | 'center'
    | 'flex-start'
    | 'flex-end'
    | 'stretch'
    | 'baseline'
    | undefined;
  justify?:
    | 'center'
    | 'flex-start'
    | 'flex-end'
    | 'space-between'
    | 'space-around'
    | 'space-evenly'
    | undefined;
  padding?: number;
  right?: number;
  horizontal?: number;
  left?: number;
  top?: number;
  vertical?: number;
  bottom?: number;
  width?: number;
}
export default class SWView extends PureComponent<Props> {
  public render() {
    return (
      <View style={styles({ ...this.props }).swView}>
        {this.props.children}
      </View>
    );
  }
}
