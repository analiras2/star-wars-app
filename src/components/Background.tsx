import React, { PureComponent } from 'react';
import colors from '../res/colors';
import Navbar from './Navbar';
import SWView from './SWView';

interface Props {
  children: Element;
  title?: string;
}
export default class Background extends PureComponent<Props> {
  public render() {
    const { title, children } = this.props;

    return (
      <SWView bgColor={colors.background} flex={1}>
        <Navbar title={title} />
        <SWView padding={20}>{children}</SWView>
      </SWView>
    );
  }
}
