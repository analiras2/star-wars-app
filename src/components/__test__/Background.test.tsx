import {shallow} from 'enzyme';
import React from 'react';
import {render} from 'react-native-testing-library';
import renderer from 'react-test-renderer';
import {Text} from 'react-native';
import Background from '../Background';
import SWView from '../SWView';

const basicView = () => (
    <Background>
       <SWView>
           <Text>Star Wars</Text>
       </SWView>
    </Background>
);

describe('Background', () => {
    const wrapper = shallow(basicView());
    const { getByText } = render(basicView());

    describe('validating Background children', () => {
        it('should render a Star Wars App', () => {
            expect(getByText(/Star Wars App/i)).toBeDefined();
        });
    });

    describe('rendering Components', () => {
        it('should render a <Navbar />', () => {
            expect(wrapper.find('Navbar')).toHaveLength(1);
        });
        it('should render a <SWView />', () => {
            expect(wrapper.find('SWView')).toHaveLength(3);
        });
    });

    it('renders SWText correctly', () => {
        const tree = renderer.create(basicView()).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
