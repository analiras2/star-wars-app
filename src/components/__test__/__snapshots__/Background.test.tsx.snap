// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background renders SWText correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#000",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 0,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "rgba(255,232,31, 0.5)",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "height": 52,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "auto",
          "color": "#FFE81F",
          "flex": 1,
          "fontFamily": "Exo-Bold",
          "fontSize": 18,
          "letterSpacing": 2,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "textAlign": "center",
        }
      }
    >
      STAR WARS APP
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "transparent",
        "flex": 0,
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "stretch",
          "backgroundColor": "transparent",
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
    >
      <Text>
        Star Wars
      </Text>
    </View>
  </View>
</View>
`;
